GVRf/Framework/framework/src/main/java/org/gearvrf/GVRScene.java

```diff
@@ -439,15 +413,6 @@ public void bindShaders(GVRSceneObject root) {
      */
     protected boolean addLight(GVRLightBase light) {
         if (light != null) {
-            int classIndex = 0;
-            for (GVRLightBase l : mLightList) {
-                if (l == light) {
-                    return false;
-                }
-                if (l.getClass().equals(light.getClass())) {
-                    ++classIndex;
-                }
-            }
             Integer lightIndex = mLightList.size();
             String name = "light" + lightIndex.toString();
             mLightList.add(light);
```

Comment:
You could be using the return value of Set.add to determine whether the light was actually added or not and return the real result to the caller.